### PR TITLE
set maximum python version to 3.10

### PR DESCRIPTION
**Pull request by cosmicBboy**
Signed-off-by: Niels Bantilan <niels.bantilan@gmail.com>

# TL;DR
Fixes https://github.com/flyteorg/flyte/issues/3313, which allows users to install flytekit on python 3.11 but then come up against errors on import.

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - ~~[ ] Unit tests added~~
 - ~~[ ] Code documentation added~~
 - ~~[ ] Any pending items have an associated Issue~~

## Complete description

This restriction was expressed softly in the `classifiers` tags but not programmatically enforced.

## Test Plan

- Create a python virtual environment on python 3.11 (either with `pyenv` or `miniconda`)
- Go to the `flytekit` root repo
- `pip install -e .` to do an editable install
- Expected outcome: you should see an error:
```
ERROR: Package 'flytekit' requires a different Python: 3.11.0 not in '<3.11,>=3.7'
```

## Tracking Issue
[https://github.com/flyteorg/flyte/issues/<number>](https://github.com/flyteorg/flyte/issues/3313)

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Review by eapolinario — DISMISSED** [review]
I wonder if it's worth removing https://github.com/flyteorg/flytekit/blob/master/setup.py#L5-L12.

**Review by eapolinario — APPROVED** [review]

**Review by eapolinario — APPROVED** [review]
